test/test_unresolved_links.cpp

```diff
@@ -261,6 +261,62 @@ TEST(Unresolved_LinkList)
     CHECK_EQUAL(stock_copy.get(3), mercedes.get_key());
 }
 
+TEST(Unresolved_NullKey)
+{
+    Group group;
+    auto table = group.add_table_with_primary_key("table", type_UUID, "_id", true);
+    auto list_col = table->add_column_list(*table, "links");
+
+    {
+        auto obj = table->create_object_with_primary_key(Mixed{}); // null is a valid key
+        auto list = obj.get_linklist(list_col);
+        list.insert(0, obj.get_key());
+        table->invalidate_object(obj.get_key());
+        CHECK_EQUAL(table->size(), 0);
+        auto unresolved_obj_key = table->get_objkey_from_primary_key(Mixed{});
+        CHECK(unresolved_obj_key.is_unresolved());
+    }
+
+    {
+        auto obj_resurrected = table->create_object_with_primary_key(Mixed{});
+        CHECK_NOT(obj_resurrected.get_key().is_unresolved());
+        CHECK_EQUAL(table->size(), 1);
```

Comment:
Should we check that backlinks work here as well? Like if there's another object in the table that also points to the null-id object in a list, that it gets resurrected correctly here?

Reply:
Good idea, I've updated the test to check this.